### PR TITLE
Set enableTcpReset of load balancer rules to true for Azure Standard Load Balancer

### LINKED ISSUE #80442: K8S Azure Cloud Provider doesn't not set enableTcpReset to true in load balancer rules for Azure Standard Load Blancer
**What happened**:
1. Create a k8s cluster backed by Azure Standard Load Balancer
2. Create a LoadBalancer type service and wait till the service is up
3. Go to Azure portal and examine the Azure Standard Load Balancer

The load balancer rule added by k8s cloud provider for the new service doesn't have a "true" value in enableTcpReset property

**What you expected to happen**:
A "true" value in the enableTcpReset property of the load balancer rule is expected.

**How to reproduce it (as minimally and precisely as possible)**:
See above.
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.13.7
- Cloud provider or hardware configuration: Azure
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by xuto2**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> 
kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
When using Azure Standard Load Balancer backed K8S cluster, previously K8S cloud provider will not set the EnableTCPReset to true in new load balancer rules added to a Standard Load Balancer. This change fixes the issue and always set the value to true as that's one of the benefits of Azure Standard Load Blancer over Basic Load Balancer.
**Which issue(s) this PR fixes**:

Fixes #80442

**Special notes for your reviewer**:
Should cherry pick to 1.13, 1.14 and 1.15.
**Does this PR introduce a user-facing change?**:

```release-note
When using Azure Standard Load Balancer backed K8S cluster, previously K8S cloud provider will not set the EnableTCPReset to true in new load balancer rules added to a Standard Load Balancer. This change fixes the issue and always set the value to true as that's one of the benefits of Azure Standard Load Blancer over Basic Load Balancer.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```
/sig azure

### CODE REVIEW

**Review by feiskyer — CHANGES_REQUESTED** [review]
Could we enable this with an annotation? see https://github.com/kubernetes/kubernetes/issues/80442#issuecomment-514016983

**Comment by xuto2** [comment]
> Could we enable this with an annotation? see [#80442 (comment)](https://github.com/kubernetes/kubernetes/issues/80442#issuecomment-514016983)
added.

**Comment by xuto2** [comment]
@feiskyer @Koderok annotation is added. PTAL.

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go [hunk 1]
```diff
@@ -899,6 +903,16 @@ func (az *Cloud) reconcileLoadBalancerRule(
...
 
+	enableTCPReset := false
+	if az.useStandardLoadBalancer() {
+		if v, ok := service.Annotations[ServiceAnnotationLoadBalancerEnableTCPReset]; ok {
+			klog.V(2).Infof("reconcileLoadBalancerRule lb name (%s) flag(%s) is set to %s", lbName, ServiceAnnotationLoadBalancerEnableTCPReset, v)
+			if strings.EqualFold(v, "true") {
+				enableTCPReset = true
+			}
+		}
+	}
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go]: Return error when basic load balancer is used here. Failing silently without errors would confuse users.

**Review by feiskyer — CHANGES_REQUESTED** [review]

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go (hunk 1 above)
Comment (xuto2) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go]: Fixed.

**Comment by feiskyer** [comment]
Thanks for the update. Could you fix the unit tests?

**Comment by xuto2** [comment]
Confirmed from PM that the enable-tcp-reset feature in Azure SLB is technically GA, the only reason that it is not publicly announced is because the Azure portal change is not ready, based on this - do you think we will need to remove this annotation one day? if so, should we do it now or we wait till Azure officially announced it? @feiskyer @Koderok

**Comment by xuto2** [comment]
closed by mistake, re-open

**Comment by xuto2** [comment]
> Thanks for the update. Could you fix the unit tests?

Fixed.

**Comment by xuto2** [comment]
> Confirmed from PM that the enable-tcp-reset feature in Azure SLB is technically GA, the only reason that it is not publicly announced is because the Azure portal change is not ready, based on this - do you think we will need to remove this annotation one day? if so, should we do it now or we wait till Azure officially announced it? @feiskyer @Koderok

A second thought on this is - we make the default behavior to true, but keep the option for user to disable it. So we reverse the annotation name to "service.beta.kubernetes.io/azure-load-balancer-disable-tcp-reset". Thoughts?

**Review by xuto2 — COMMENTED** [review]
Pending further design discussion.

**Comment by feiskyer** [comment]
> So we reverse the annotation name to "service.beta.kubernetes.io/azure-load-balancer-disable-tcp-reset". Thoughts?

LGTM.

**Comment by xuto2** [comment]
closing as this is a bug but not a feature.

**Pull request closed without merging**